src/main/java/bio/terra/app/controller/SearchApiController.java

```diff
@@ -80,6 +85,14 @@ private AuthenticatedUserRequest getAuthenticatedInfo() {
     return authenticatedUserRequestFactory.from(request);
   }
 
+  @Override
+  public ResponseEntity<Object> enumerateSnapshotSearch() {
+    List<UUID> authorizedResources =
+        iamService.listAuthorizedResources(getAuthenticatedInfo(), IamResourceType.DATASNAPSHOT);
+    Map<UUID, String> metadata = searchService.enumerateSnapshotSearch(authorizedResources);
+    return ResponseEntity.status(HttpStatus.OK).body(metadata);
```

Comment:
It's a minor thing but I think `.ok()` is a bit more convenient. Also I think we only want to return the `values()` since each metadata blob contains the uuid.
```suggestion
    return ResponseEntity.ok(metadata.values());
```